goose.go

```diff
@@ -1346,7 +1351,7 @@ func (ctx Ctx) sliceRangeStmt(s *ast.RangeStmt) coq.Expr {
 		Val:   ctx.identBinder(val),
 		Slice: ctx.expr(s.X),
 		Ty:    ctx.coqTypeOfType(s.X, sliceElem(ctx.typeOf(s.X))),
-		Body:  ctx.blockStmt(s.Body, loopVar),
+		Body:  ctx.blockStmt(s.Body, ExprValLocal),
```

Comment:
There's an easy path to supporting control effects in slice (and map) range statements now: we translate this with `ExprValLoop`, and adjust `coq.SliceLoopExpr` generation such that the loop body should return `Break` or `Continue` to tell the iterator function what to do.